### ISSUE TITLE
Ingestion request: TAL

### ISSUE THREAD

**Issue by mcandito** [acl-org/acl-anthology#2339]
Hello,

This follows a request I originally made by email to Matt (on December 13 2022),
on behalf of the editorial board of the TAL journal ( https://www.atala.org/revuetal )
which is the journal of the French association for computational linguistics (ATALA)

Matt agreed for the ingestion of the past and future articles.
Thank you very much, this is great.

The venue name could be "tal".

I am currently preparing the files in aclpub format for all the past volumes
that are electronically available, namely the last 16 volumes.
There is one volume per year, and each volume contains 3 issues.

Best,
Marie Candito

**Comment by mcandito** [comment]
I prepare one data directory for each volume (= each year).


data/
  tal-2021-62-1/     (= first issue of volume 62)
     proceedings/
         meta
         cdrom/
         ...

  tal-2021-62-2/   (= second issue of volume 62)
     proceedings/
         ...

Could you check if I'm right

I tried to look at examples of other non-acl journals in the anthology.
For the IJCNLP journal,
the url keeps the year and the number, but not the volume
(cf. https://aclanthology.org/2021.ijclclp-1.0.pdf for the volume 26, number 1)

I would rather keep the year, the volume and the issue number.
So for instance for the year 2021, Volume 62, issue number 2,
the bib entry for whole number 2 would be:

@proceedings{tal-2021-62-2,
  title = {{Traitement Automatique des Langues}, {V}olume 62, Numéro 2 : {Nouvelles applications du TAL [New applications in NLP]}},
  editor = {Damnati, Géraldine and Inkpen, Diana},
  publisher = {{ATALA (Association pour le Traitement Automatique des Langues)}},
  year = "2021",
  address = "France",
  url = {https://aclanthology.org/2021.tal-62-2.0},
}

and the meta file:

abbrev Tal
title Nouvelles applications en TAL [New applications in NLP]
booktitle Traitement Automatique des Langues 2021 Volume 62 Numéro 2
shortbooktitle TAL 2021 62:2
year 2021
volume 62-2
chairs Géraldine Damnati
chairs Diana Inkpen
location France
publisher ATALA (Association pour le Traitement Automatique des Langues)

**Comment by mjpost** [comment]
Thanks. Let's do one of these to get all the details right and then you can move to the rest? Please share and then @xinru1414 will process.

Are the proceedings always and completely in French?

**Comment by mjpost** [comment]
Thanks, Marie. I asked about language because we can annotate papers in the Anthology with their main language. That would be easy if they were all French. Since they're not, I wonder if you have the annotation made at the paper level?

Ping @xinru1414 for the ingest.

**Comment by mcandito** [comment]
Hi,
Yes we have the annotation of the language at the paper level,
currently my script adds language="French" in the bib entries
tell me if you would rather have this info in some other format
Thanks

**Comment by mcandito** [comment]
Hi @xinru1414 
To check if the format is ok,
could you try with these two volumes, for the new venue name "tal":
http://www.linguist.univ-paris-diderot.fr/~mcandito/divers/2018.zip
http://www.linguist.univ-paris-diderot.fr/~mcandito/divers/2011.zip

For some of the volumes, there is no pdf for a whole issue, only pdfs for each paper.
So for instance while there is a 2018.tal-59-1.pdf file, there is no 2011.tal-52-3.pdf

Thank you

**Comment by mcandito** [comment]
Hi @xinru1414,
did you have a look at the zip files?
Thanks
Marie

**Comment by mcandito** [comment]
Hi,
@xinru1414 I think Matt intended to try like this 
(cf. his previous answer "Let's do one of these to get all the details right")
Thanks

**Comment by mcandito** [comment]
Hi @xinru1414,
can you try the ingestion of the two archives as they are currently?
If I understood well, Matt meant to try and see.
Thanks

**Comment by mjpost** [comment]
@xinru1414 yes, please do the ingest of just one volume, and we'll see how it looks.

**Comment by mcandito** [comment]
Hi @xinru1414,

I confirm that my first request was to try to keep the year, and both volume and issue number.
It seemed unclear whether this was feasible or not, and I thought you had to try it.
Do you confirm that this is not feasible ?

If this is not possible, then combining all issues of a volume is not an option because there are different editors for each issue. 
I would thus prefer to use the same solution as IJCNLP (cf. above),
namely keep the year and issue number only.

Tell me if I should modify my script accordingly and send you another archive for test
Thanks a lot for this work!

**Comment by mjpost** [comment]
Oh, I see the issue. Look at CL to see how this is done. The problem is overloaded terminology. In our lingo, the journal's "issue" becomes our `<volume>` tag. The journal's volume is represented only in the title. See https://github.com/acl-org/acl-anthology/blob/master/data/xml/2021.cl.xml for an example. Is this helpful?

**Comment by mcandito** [comment]
Hi, 
I've modified my script 
so that the "volume" (in acl lingo) is now 1 (or 2 or 3) and not "59-1" as previously envisaged.

@xinru1414 can you try with these new ones:
http://www.linguist.univ-paris-diderot.fr/~mcandito/tmp/2018-new.zip
http://www.linguist.univ-paris-diderot.fr/~mcandito/tmp/2011-new.zip

The only difference between the two is that in 2011, there is no pdf for the full issues
(hence no file 2011.tal-1.pdf for instance)
Thank you

**Comment by mjpost** [comment]
Hi @mcandito, sorry, I meant for @xinru1414 to do this. But thanks for your help.

@xinru1414, looking into this, it seems that our journal-handling is hard-coded, that is, we have [a list of journals in data.py](https://github.com/acl-org/acl-anthology/blob/dae3430d53085bfea11491b005f4907375702dbe/bin/anthology/data.py#L72), and when we generate the bibtex, we [parse the booktitle](https://github.com/acl-org/acl-anthology/blob/dae3430d53085bfea11491b005f4907375702dbe/bin/anthology/volumes.py#L135) to figure out volume and issue numbers.

There are a number of issues with this; for one, we have other journals (e.g., [LILT](https://aclanthology.org/venues/lilt/)) that are not currently processed as such. Moreover, we should move this out of the code and into the XML, and adapt the code accordingly.

The short-term fix is just to use the issue in the `<volume>` tag as @mcandito is doing above, copying CL, and make sure the parsing code correctly grabs the volume and issue for the bibtex. Immediately—and even together with this, if you have the bandwidth—we should add explicit `<volume-number>` and `<issue>` tags (names might change) to the XML, moving it out of the code, for all journals.

**Comment by mcandito** [comment]
Hi @xinru1414,

I've studied the preview of the tal ingestion #2384

I've seen a bug on my side (mismatch of pdfs for the varia issues)

Could you please consider these two debugged archives and redo the ingestion test??
http://www.linguist.univ-paris-diderot.fr/~mcandito/tmp/2011.zip
http://www.linguist.univ-paris-diderot.fr/~mcandito/tmp/2018.zip

Then I am lost about the procedure,
who is performing the "approving review" ?
am I supposed to do anything?

Thanks

**Comment by xinru1414** [comment]
@mcandito Once the ingestion is done, please take a look at the preview page and let us know if there are any issues. You don't have to do anything else. We will review, fix and merge the PR.

**Comment by mcandito** [comment]
Hi @xinru1414 , it seems I'm late, cf. I can't see the TAL preview anymore, nor the final view on aclanthology.org
Thanks

**Comment by mcandito** [comment]
@xinru1414 I can see this page https://aclanthology.org/venues/tal/ (thank you!!)
but TAL does not seem available from the main page

**Comment by mcandito** [comment]
Hi @xinru1414, 
I noticed there is no page for full issues
for instance
there is no https://aclanthology.org/2018.tal-1.0/
whereas this is the case e.g. for the IJCLCLP journal
https://aclanthology.org/2021.ijclclp-1.0/

Could you have a look ?
I am preparing the ingestion requests for the other volumes of the TAL journal
Thanks

**Comment by mcandito** [comment]
Hi @mjpost 
From the start I took IJCLCLP as example, being the only non-ACL journal I could find on the anthology
and their .0 pages do concern the whole issues, and point to the pdf of the whole issue
https://aclanthology.org/2021.ijclclp-1.0/
and don't seem to have separate frontmatters

Do you have the possibility to send me one of their zip so that I can mimick the same structure?

Indeed, in the zips I provided, I followed the structure recommended for the ACLPUB format

Maybe I should just provide the pdf of the whole issue
in the zip as

tal-2018-2 / proceedings / cdrom / pdf / 2018.tal-2.0.pdf

instead of what I currently provide

tal-2018-2 / proceedings /cdrom / pdf / 2018.tal-2.pdf

(actually I realize anyway in ACLPUB the pdf for the whole issue is supposed to be at
tal-2018-2 / proceedings /cdrom / 2018.tal-2.pdf
my mistake)

?

**Comment by mcandito** [comment]
@mjpost 
Actually the important thing is not so much to have access to the pdfs of whole issues
but rather to have bibtex entries for the whole issues.
It seems that because I didn't provide .0.pdf files 
then the .0.bib I provided (entries for whole issues) were not taken into account either.

Cf. the first entry of 
https://aclanthology.org/volumes/2021.ijclclp-1.bib
is for the whole issue

whereas it is missing in
https://aclanthology.org/volumes/2018.tal-1.bib

Moreover, this is further complicated by the fact 
I have the pdfs of whole issues for the issues posterior to 2016 only.
That's why I sent you both 2018 and 2011, as a test for volumes with and without pdf for whole issue.

Is it possible to wait until we sorted this out before publishing on the front page?

Thanks for reading so far...!

**Comment by mjpost** [comment]
Hi @mcandito (CC: @acl-anthology)—it seems we did not create a `frontmatter` block. This is only created if there is a "front matter" PDF. @acl-anthology, can you edit our ingestion scripts to *always* create the `<frontmatter>` block, even if there isn't a PDF? In the meantime, I manually added this to a new PR (#2412).

**Comment by mcandito** [comment]
Hi @mjpost, thanks for the manual work

"can you edit our ingestion scripts to always create the <frontmatter> block"
=> there does not seem to be a <frontmatter> block in the ACLPUB format I provide you,
can you tell me what precise feature triggers the creation of the <frontmatter> block in your scripts ??

Is it the existence of a XXX.0.pdf file in 
data / XXX / proceedings / cdrom / pdf / XXX.0.pdf ?
Thanks
